File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.py
def solution(arr, k):
    answer = []
    for a in arr:
        if a not in answer:
            answer.append(a)
            if len(answer) >= k:
                break
    
    if len(answer) < k:
        for i in range(k-len(answer)):
            answer.append(-1)
        
    return answer

File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int k) {        
        Deque<Integer> answer = new ArrayDeque<Integer>();
        
        for (int num : arr) {
            if (!answer.contains(num) && answer.size() < k) {
                answer.offer(num);   
            }
        }
        
        int len = answer.size();
        for (int i = 0; i < k - len; i++) {
            answer.offer(-1);
        }
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}